### ISSUE TITLE
Verbose console repeats task names many times in parallel build

### ISSUE THREAD

**Issue by ilya-g** [gradle/gradle#3456]
I've noticed that verbose console tends to repeat task names without any output during parallel builds. Here is excerpt from the output when it happens: https://pastebin.com/8rGarXF5
I've magnified the corresponding part of the build scan timeline: https://scans.gradle.com/s/umo3mre7vjf3i/timeline#end=1510517317022&start=1510517291452

### Expected Behavior
I expect that task name should be printed only when a task outputs something or when it completes.

### Current Behavior
Task names are repeated with no output several times for each task

### Steps to Reproduce
Run parallel build with `--console verbose`

 * Build scan URL: https://scans.gradle.com/s/umo3mre7vjf3i

**Comment by blindpirate** [comment]
Thanks for reporting. Actually, AFAIK, the task names do be printed only when task outputs something. Could you provide us a sample project to reproduce this issue? It would help a lot.

**Comment by ilya-g** [comment]
I think any project with long-running parallel tasks will do, but in case if you struggle to find one, you can reproduce it on the Kotlin build, of which I took that scan. The instructions are here https://github.com/jetbrains/kotlin#building

**Comment by oehme** [comment]
This probably happens because we force output every now and then to clear the buffers. However, nothing should happen if the buffers are empty.

**Comment by stmlange** [comment]
Hi,
I bumped into this by accident as well and I noticed that I can reproduce the issue for gradle 4.6, but for gradle 4.7 this behaviour seems to be fixed.

For reference I have created a small sample that is using two sleeper tasks and the worker api to have them run in parallel. When using gralde 4.6 in the wrapper-properties one can reproduce the outlined problem, with gradle 4.7 it "goes away".

[issue-3456.zip](https://github.com/gradle/gradle/files/1944554/issue-3456.zip)
